vm/systemSmartContracts/auction.go

```diff
@@ -412,11 +418,43 @@ func (s *stakingAuctionSC) setConfig(args *vmcommon.ContractCallInput) vmcommon.
 		UnJailPrice:   big.NewInt(0).SetBytes(args.Arguments[5]),
 	}
 
+	if auctionConfig.MinStakeValue.Cmp(zero) <= 0 {
+		retMessage := fmt.Errorf("%w, value is %v", vm.ErrInvalidMinStakeValue, auctionConfig.MinStakeValue).Error()
+		s.eei.AddReturnMessage(retMessage)
+		return vmcommon.UserError
+	}
+	if auctionConfig.NumNodes < 1 {
+		retMessage := fmt.Errorf("%w, value is %v", vm.ErrInvalidMinNumberOfNodes, auctionConfig.NumNodes).Error()
+		s.eei.AddReturnMessage(retMessage)
+		return vmcommon.UserError
+	}
+	if auctionConfig.TotalSupply.Cmp(zero) <= 0 {
+		retMessage := fmt.Errorf("%w, value is %v", vm.ErrInvalidGenesisTotalSupply, auctionConfig.TotalSupply).Error()
+		s.eei.AddReturnMessage(retMessage)
+		return vmcommon.UserError
+	}
+	if auctionConfig.MinStep.Cmp(zero) <= 0 {
+		retMessage := fmt.Errorf("%w, value is %v", vm.ErrInvalidMinStepValue, auctionConfig.MinStep).Error()
+		s.eei.AddReturnMessage(retMessage)
+		return vmcommon.UserError
+	}
+	if auctionConfig.NodePrice.Cmp(zero) <= 0 {
+		retMessage := fmt.Errorf("%w, value is %v", vm.ErrInvalidNodePrice, auctionConfig.NodePrice).Error()
+		s.eei.AddReturnMessage(retMessage)
+		return vmcommon.UserError
+	}
+	if auctionConfig.UnJailPrice.Cmp(zero) <= 0 {
+		retMessage := fmt.Errorf("%w, value is %v", vm.ErrInvalidUnJailCost, auctionConfig.UnJailPrice).Error()
+		s.eei.AddReturnMessage(retMessage)
+		return vmcommon.UserError
+	}
+
 	configData, err := s.marshalizer.Marshal(auctionConfig)
 	if err != nil {
 		s.eei.AddReturnMessage("setConfig marshal auctionConfig error")
 		return vmcommon.UserError
 	}
+	// TODO: check if updating older epochs should be allowed
```

Comment:
no need for this TODO - it is still undecided how setConfig is used, or whether it will be used.

Reply:
ok, removed